app/routes.py

```diff
@@ -1,2 +1,245 @@
+                goals_response.append({
+                    "id": goal.goal_id,
+                    "title": goal.title
+                })
+            sort_by_title = request.args.get("sort")
+            if sort_by_title:
+                if sort_by_title == "asc":
+                    goals_response = sorted(goals_response, key = lambda i: i['title'])
+                if sort_by_title == "desc":
+                    goals_response = sorted(goals_response, key = lambda i: i['title'],reverse=True)
+            return jsonify(goals_response), 200
+
+        elif request.method == "POST":
+            request_body = request.get_json()
+            if 'title' not in request_body:
+                return {"details": "Invalid data"}, 400
+            new_goal = Goal(title=request_body["title"])
+
+            db.session.add(new_goal)
+            db.session.commit()
+
+        return {
+            "goal": {
+                "id": new_goal.goal_id,
+                "title": new_goal.title
+            }
+        }, 201
+
+@goals_bp.route("/<goal_id>", methods=["GET", "DELETE", "PUT"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        return {
+            "goal":{
+                "id": goal.goal_id,
+                "title": goal.title
+            }
+        }
+    elif request.method == "DELETE":
+        message = {"details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"}
+        db.session.delete(goal)
+        db.session.commit()
+        return make_response(message)
+    elif request.method == "PUT":
+        form_data = request.get_json()
+
+        goal.title = form_data["title"]
+
+        db.session.commit()
+
+        return make_response({
+                "goal": {
+                    "id": goal.goal_id,
+                    "title": goal.title
+                }
+        })
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST", "GET"])
+def handle_goal_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        tasks = Task.query.filter(Task.goal_id == goal_id)
+        results = []
+        for task in tasks:
+            results.append({
+                "id": task.task_id,
+                "goal_id": task.goal_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": True if task.completed_at else False
+            })
+
+        return make_response(
+                {
+                    "id": goal.goal_id,
+                    "title": goal.title,
+                    "tasks": results
+                }, 200)
+    elif request.method == "POST":
+        form_data = request.get_json()
+        task_ids = form_data['task_ids']
+
+        for id in task_ids:
+            task = Task.query.get(id)
+            task.goal_id = goal_id
```

Comment:
👍 

Yep. We need to retrieve each task to be able to associate it to the goal.

But what would happen if the goal previously had some tasks set. Do we want to add the new tasks to the existing tasks? Do we want to replace them and sever any prior task → goal relationships? What behavior is implemented here?